mls-rs/src/group/padding.rs

```diff
@@ -31,6 +35,24 @@ impl PaddingMode {
 
                 (content_size | (blind - 1)) + 1
             }
+            PaddingMode::Padme => {
+                // Prevents log2(0), which is undefined.
+                if content_size < 2 {
+                    return content_size;
+                }
+
+                // E <- floor(log2(L))
+                // S <- floor(log2(E)) + 1
+                // z <- E - S
+                // m <- (1 << z) - 1
+                // len' <- (L + m) & ~m
+
+                let e: u32 = content_size.ilog2(); // l’s floating-point exponent
+                let s: u32 = e.ilog2() + 1; // number of bits to represent e
+                let num_zero_bits: u32 = e - s; // number of low bits to set to 0
+                let bitmask: usize = (1 << num_zero_bits) - 1; // create a bitmask of 1s
```

Comment:
```suggestion
                let num_one_bits: u32 = e - s; // number of low bits to set to 1
                let bitmask: usize = (1 << num_one_bits) - 1; // create a bitmask of 1s
```
Maybe I'm confused, but we're fixing to 1's to increase the length?

Reply:
Can be a follow-up if needed/

Reply:
Ye, the usage here is a little convoluted wrt the variable naming. Note that `bitmask` is ultimately used as `!bitmask`. Instead of setting `E = exponent` number of low bits to 0 (which is equivalent to padding to powers of 2), we set `E-S` low bits to 0 (which allows more granular padding, though more leakage of size). Before setting `E-S` low bits to 0, we add `E-S` to the value to round up if needed.

We could rename some variables to make this clearer, though it'll then diverge from the reference pseudocode in the paper. LMK what you prefer.